docs/autoredraw.md

```diff
@@ -45,6 +45,61 @@ function doSomething(e) {
 m.mount(document.body, MyComponent)
 ```
 
+Mithril.js also redraws once promise returned from event handler resolves. It makes it easy to work with async code.
+
+```javascript
+var count = 0
+
+async function incrementAfterDelay() {
+	await new Promise(
+		(resolve) => setTimeout(resolve, 500)
+	)
+
+	count++
+}
+
+var DelayedCounterComponent = {
+	view: function() {
+		return m("button", {
+			onclick: incrementAfterDelay,
+		}, `Add one (${count})`)
+	}
+}
+
+m.mount(document.body, DelayedCounterComponent)
+```
+
+You still have to run `m.redraw()` manually in case you need redraw before whole promise chain is resolved.
+
+```javascript
+var count = 0
+
+async function incrementAfterDelay() {
+	await new Promise(
+		(resolve) => setTimeout(resolve, 500)
+	)
+
+	count++
+
+	m.redraw()
+}
+
+async function complexOperation() {
+	await incrementAfterDelay()
+	await incrementAfterDelay()
+	await incrementAfterDelay()
+}
+
+var ComplexDelayedCounterComponent = {
+	view: function() {
+		return m("button", {
+			onclick: complexOperation,
+		}, `Add 3 (${count})`)
+	}
+}
+
+m.mount(document.body, ComplexDelayedCounterComponent)
```

Comment:
It would be great if the example was a bit more practical ie making a network request and changing state on success/fail in the event handler.

Also I think it would be great to inline the async handler ie 

```
m('button', {
  async onclick(){ 
    await m.request(...)
    // etc
  }
})
```

It makes it clear that mithril is natively supporting async methods in the hyperscript.